### ISSUE TITLE
Regression: [macOS] Build: Yuzu no longer builds on Mac

### ISSUE THREAD

**Issue by shinra-electric** [yuzu-emu/yuzu#11761]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

N/A

### Description of Issue

Building failed with the following: 
```
[1741/2271] Building CXX object src/co...core.dir/hle/service/caps/caps_a.cpp.o
FAILED: src/core/CMakeFiles/core.dir/hle/service/caps/caps_a.cpp.o 
```

This is likely due to changes introduced in commit 8347e5c which touches `caps_a.cpp`

### Expected Behavior

The last time I built successfully was on the 8th of October, so the change must have happened between then and now.

I should bisect the commits to find the exact one, but I'm a bit short on time and it seems likely that 8347e5c was the culprit.

### Reproduction Steps

Build Yuzu for macOS 

### Log File

[Yuzu terminal.txt](https://github.com/yuzu-emu/yuzu/files/12882140/Yuzu.terminal.txt)

### System Configuration

CPU: M1 Pro
GPU/Driver: M1 Pro
RAM: 16GB
OS: 14.0 Sonoma

**Comment by german77** [comment]
We need a CI for mac. Fixing the issue soon

**Comment by shinra-electric** [comment]
The fix in #11765 did allow the build to get further, but it still fails further along: 

```
[2254/2275] Building CXX object src/yuzu/CMakeFiles/yuzu.dir/main.cpp.o
FAILED: src/yuzu/CMakeFiles/yuzu.dir/main.cpp.o 
```

[Yuzu Terminal.txt](https://github.com/yuzu-emu/yuzu/files/12890556/Yuzu.Terminal.txt)

**Comment by shinra-electric** [comment]
Fixed by #11920
